samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -42,8 +42,8 @@ func (a PetApi) AddPet (body Pet) (APIResponse, error) {
   path := a.Configuration.BasePath + "/pet"
 
   // verify the required parameter 'body' is set
-  if &body == nil {
-      return *NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter 'body' when calling PetApi->AddPet")
+  if reflect.DeepEqual((Pet{}), body) {
```

Comment:
@neilotoole I think I agree with you, validating complex object in Go does not make sense. this is the shortcoming of Go.(not "supporting" nullable object, but on the other hand, Go wouldn't compile if i put `nil` as parameter, which is a good thing)

I have removed this validation in the template as well as the test.